### DISCUSSION TITLE
How to Mock useFormContext inside child component

### DISCUSSION THREAD

**Discussion by auppal80** [react-hook-form/react-hook-form#5781] [Q&A]
Hello Everyone

I have been trying to mock useFormContext  inside a child component for last one day but no success.  child component has this line of code

```tsx
const { register } = useFormContext<CustomerProfileFormData>();
```

I am trying to write test for ths child component. How can I mock useFormContext<CustomerProfileFormData>();

I tried this
```tsx
 const mockControl = {};
        const mockErrors: string[] = [];
        jest.mock('react-hook-form', () => ({
            useFormContext: () =>
                jest.fn(() => {
                    control: mockControl;
                    errors: mockErrors;
                    register: jest.fn();
                }),
        }));
        renderWithProviders(<AddressDetails isEdit={false} />);
```
but I get this error 
> TypeError: Cannot destructure property 'register' of '(0 , _reactHookForm.useFormContext)(...)' as it is null.Jest. 
Can someone please help me.

**Comment by bluebill1049** [comment]
hi there, don't mock it. instead, bring `FormProvider` and test.

have a look at here: https://github.com/react-hook-form/react-hook-form/blob/master/src/__tests__/useFormContext.test.tsx
